### DISCUSSION TITLE
How to Implement Persistent Summarized Chat History in PromptFlow?

### DISCUSSION THREAD

**Discussion by lucaordronneau** [microsoft/promptflow#3519] [Q&A] [answered]
Hello, I am interested in exploring advanced memory management techniques in chat mode (Conversation Summary).

Currently, `chat_history` is the only persistent variable and it cannot be modified directly and the primary method to reduce the number of tokens seems to be filtering this `chat_history`.

I am looking for a way to **modify** `chat_history` in a manner that its modifications **persist** across different instances of the chat.
Specifically, I am interested in modifying `chat_history` to pass **a summary** of the history.

Is it possible to achieve this with the current version of the library? If not, is there an alternative approach to implement this feature?

**Comment by D-W-** [comment, marked as answer]
Hi @lucaordronneau , currently we don't have such feature. A workaround would be use LangChain's [ConversationSummaryBufferMemory](https://python.langchain.com/v0.1/docs/modules/memory/types/summary_buffer/) and wrap it into a flex flow. We have a sample on how to use LangChain in prompt flow: https://github.com/microsoft/promptflow/tree/main/examples/flex-flows/eval-criteria-with-langchain
